apps/ejabberd/src/mod_keystore.erl

```diff
@@ -121,7 +121,11 @@ create_keystore_ets() ->
             ok
     end.
 
-maybe_add_heir(EjdSupPid, _Self, BaseOpts) when is_pid(EjdSupPid) ->
+%% In tests or when module is started in run-time, we need to set heir to the
+%% ETS table, otherwise it will be destroy when the creator's process finishes.
+%% When started normally during node start up, self() =:= EjdSupPid and there
+%% is no need for setting heir
+maybe_add_heir(EjdSupPid, EjdSupPid, BaseOpts) ->
     [{heir, EjdSupPid, testing} | BaseOpts];
```

Comment:
I'm a bit confused. When calling `maybe_add_heir` function with the same `EjdSupPid` as a `_Self` pid we should avoid setting the heir (as it's clearly explained in comment). Didn't you mean:
```
maybe_add_heir(EjdSupPid, EjdSupPid, BaseOpts) when is_pid(EjdSupPid) ->
     BaseOpts;
maybe_add_heir(EjdSupPid, _Self, BaseOpts) when is_pid(EjdSupPid) ->
      [{heir, EjdSupPid, testing} | BaseOpts];
maybe_add_heir(_, _, BaseOpts) ->
      BaseOpts.
```